src/components/add-to-playlist-drawer/AddToPlaylistDrawer.tsx

```diff
@@ -0,0 +1,157 @@
+  addedToast: 'Added To Playlist!',
+  createdToast: 'Playlist Created!'
+}
+
+const createStyles = (themeColors: ThemeColors) =>
+  StyleSheet.create({
+    cardContainer: {
+      display: 'flex',
+      flexDirection: 'row',
+      flexWrap: 'wrap',
+      justifyContent: 'flex-start',
+      padding: 12,
+      paddingTop: 24
+    },
+    card: {
+      flex: 1,
+      flexBasis: '40%',
+      maxWidth: '50%',
+      marginBottom: 8
+    },
+    newPlaylistButton: {
+      backgroundColor: themeColors.white,
+      height: 48,
+      width: 256
+    },
+    newPlaylistButtonContainer: {
+      backgroundColor: 'transparent'
+    },
+    newPlaylistButtonText: {
+      color: themeColors.neutral
+    }
+  })
+
+export const AddToPlaylistDrawer = () => {
+  const styles = useThemedStyles(createStyles)
+  const dispatchWeb = useDispatchWeb()
+  const pushRouteWeb = usePushRouteWeb()
+  const isOpen = useSelectorWeb(getIsOpen)
+  const trackId = useSelectorWeb(getTrackId)
+  const trackTitle = useSelectorWeb(getTrackTitle)
+  const user = useSelectorWeb(getAccountWithOwnPlaylists)
+  const [isDrawerGestureSupported, setIsDrawerGestureSupported] = useState(true)
+
+  const handleClose = useCallback(() => {
+    dispatchWeb(close())
+  }, [dispatchWeb])
+
+  if (!user || !trackId || !trackTitle) return null
+  const userPlaylists = user.playlists ?? []
+
+  const addToNewPlaylist = () => {
+    const metadata = newCollectionMetadata({
+      playlist_name: trackTitle,
+      is_private: false
+    })
+    const tempId = `${Date.now()}`
+    dispatchWeb(
+      createPlaylist(tempId, metadata, CreatePlaylistSource.FROM_TRACK, trackId)
+    )
+    dispatchWeb(addTrackToPlaylist(trackId!, tempId))
+    toast({ content: messages.createdToast })
+    // TODO: Update the fromPage location. Need to figure out how to get page that the user came from
+    pushRouteWeb(playlistPage(user.handle, trackTitle, tempId), FEED_PAGE)
+    handleClose()
+  }
+
+  const handleScrollEnd = (e: NativeSyntheticEvent<NativeScrollEvent>) => {
+    const { y } = e.nativeEvent.contentOffset
+
+    if (isDrawerGestureSupported && y > 0) {
+      setIsDrawerGestureSupported(false)
+    } else if (!isDrawerGestureSupported && y <= 0) {
+      setIsDrawerGestureSupported(true)
+    }
+  }
+
+  return (
+    <Drawer
+      isOpen={isOpen}
+      onClose={handleClose}
+      isFullscreen
+      isGestureSupported={isDrawerGestureSupported}
+      title={messages.title}
+    >
+      <ScrollView onScrollEndDrag={handleScrollEnd}>
+        <Shadow
+          offset={[0, 1]}
+          containerViewStyle={{ alignSelf: 'center' }}
+          distance={2}
+          startColor='rgba(133,129,153,0.16)'
+        >
+          <Button
+            title='Create New Playlist'
+            onPress={addToNewPlaylist}
+            type={ButtonType.COMMON}
+          />
+        </Shadow>
+        <View style={styles.cardContainer}>
+          {userPlaylists.map((playlist, idx) => (
```

Comment:
we interested in doing a FlatList here, or were there scroll issues?

Reply:
oh yeah nice, we have a cool helper now, i think you can use `CardList` component!